app/views/homepage/_list_item_with_distance.haml

```diff
@@ -0,0 +1,70 @@
+- frontpage_fragment_cache("list_item", listing) do
+  .browsing-list-item
+    .browsing-list-item__image-container
+      - if listing.listing_images.size > 0
+        = link_to listing_path(listing.url), :class => "browsing-list-item__image-desktop" do
+          = image_tag listing.listing_images.first[:small_3x2], {:alt => listed_listing_title(listing), :class => "home-list-image"}
+      - if listing.listing_images.size > 0
+        = link_to listing_path(listing.url), :class => "browsing-list-item__image-mobile" do
+          = image_tag listing.listing_images.first[:thumb], {:alt => listed_listing_title(listing), :class => "home-list-image"}
+    .browsing-list-item__details
+      .browsing-list-item__details-right
+        .browsing-list-item__price
+          - if listing.price
+            .browsing-list-item__price-value
+              = humanized_money_with_symbol(listing.price).upcase
+            - price_text = nil
+            - if listing.quantity.present?
+              - price_text = t("listings.form.price.per") + " " + listing.quantity
+            - elsif listing.unit_type
+              - price_text = price_quantity_per_unit(listing)
+
+            - if price_text.present?
+              .browsing-list-item__price-quantity{:title => price_text}
+                = price_text
+          - else
+            .browsing-list-item__listing-shape-value
+              = shape_name(listing)
+
+      .browse-list-item__details-container
+        %h2.browsing-list-item__title
+          = link_to listing_path(listing.url) do
+            = listing.title
+            - if @current_community.show_category_in_listing_list
+              %a.browsing-list-item__share-type-link{:href => root_path(:transaction_type => shape_name_map[listing.listing_shape_id], :view => :list)}
+                = icon_tag(listing.icon_name, ["icon-fix"])
+                = shape_name(listing)
+        - distance = Maybe(listing.distance).or_else(nil)
```

Comment:
Maybe is not necessary here. `Maybe(listing.distance).or_else(nil)` is equal to just `listing.distance`